al-khaser/Anti Debug/NtQueryObject_AllTypesInformation.cpp

```diff
@@ -10,6 +10,8 @@ locate "DebugObject". Todo: Support for Win10
 
 BOOL NtQueryObject_ObjectAllTypesInformation()
 {
+	//NOTE this check is unreliable, a debugger present on the system doesn't mean it's attached to you
+

```

Comment:
That's not true. You explicitly create a debug object in the other check (so basically check for hooks). I modified that function to pass if one or more debug objects are found (so it will pass if there is a debugger on the system, but fail if the function is clearly hooked).